### DISCUSSION TITLE
Replacement for QuantConvTranspose2d

### DISCUSSION THREAD

**Discussion by hleblevec** [Xilinx/finn#505] [Q&A] [answered]
Hi,

I am trying to implement a UNet using FINN, and need to use transposed convolutions for upsampling purposes. I noticed when trying to export the Brevitas model that the _QuantConvTranspose2d_ layer was not supported for export. I assume this on the roadmap to expand FINN's features. 
As for now, do you have any suggestion of replacement for those layers that would be supported by FINN?
Would it work if I used _torch.nn.functional.interpolate_ instead?

Regards,

Hugo

**Comment by hleblevec** [comment, marked as answer]
Hi,
So, for the record, you can replace a ConvTranspose2d layer by an interpolate followed by a regular Conv2d layer at the expense of a small loss in accuracy. This then converts into a QuantUpsample and QuantConv2d in Brevitas, which are compatible with FINN export.
